packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -355,33 +355,40 @@ IOSink? _openSink(String? output) {
   return sink;
 }
 
-/// A generator that will write code to a sink based on the contents of [PigeonOptions].
-abstract class Generator {
-  /// Returns an [IOSink] instance to be written to if the [Generator] should
-  /// generate.  If it returns `null`, the [Generator] will be skipped.
+/// An adapter that will call a generator to write code to a sink
+/// based on the contents of [PigeonOptions].
+abstract class Adapter {
```

Comment:
`GeneratorAdapter` would be a better name; "Adapter" could be adapting anything.

Also, can this be private?

Also, this is a breaking change, so needs to be versioned as such. Everything in Pigeon is directly in `lib`, so everything that doesn't start with `_` is considered public.

Reply:
This is currently public for testing purposes. I'd be open to making adjustments to change this though. @gaaclarke mentioned adding the @public_for_testing tag on these classes.